staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -836,9 +853,11 @@ loop:
 					utilruntime.HandleError(fmt.Errorf("%s: unable to add watch event object (%#v) to store: %v", name, event.Object, err))
 				}
 			case watch.Modified:
-				err := store.Update(event.Object)
-				if err != nil {
-					utilruntime.HandleError(fmt.Errorf("%s: unable to update watch event object (%#v) to store: %v", name, event.Object, err))
+				if !immutable {
+					err := store.Update(event.Object)
+					if err != nil {
+						utilruntime.HandleError(fmt.Errorf("%s: unable to update watch event object (%#v) to store: %v", name, event.Object, err))
+					}
```

Comment:
Performance: This does not disable the watch for changes on the kubernetes api server, only skips handling on the "client". The main reason for stopping the watch was to relieve the apiserver of the load of watchers for immutable secrets.
With this change this load returns to the apiserver. Let's discuss our options synchronously